plugins/entity-feedback/src/components/FeedbackResponseDialog/FeedbackResponseDialog.tsx

```diff
@@ -81,13 +87,41 @@ export const FeedbackResponseDialog = (props: FeedbackResponseDialogProps) => {
   const classes = useStyles();
   const errorApi = useApi(errorApiRef);
   const feedbackApi = useApi(entityFeedbackApiRef);
+  const alertApi = useApi(alertApiRef);
   const [responseSelections, setResponseSelections] = useState(
     Object.fromEntries(feedbackDialogResponses.map(r => [r.id, false])),
   );
   const [comments, setComments] = useState('');
   const [consent, setConsent] = useState(true);
+  const requireComments = feedbackDialogResponses
+    .filter(r => r.mustComment)
+    .map(r => r.id);
+
+  const isLinkedBoxChecked = () => {
+    const checkedBoxes = Object.keys(responseSelections).filter(
+      id => responseSelections[id],
+    );
+    return checkedBoxes.some(id => requireComments.includes(id));
+  };
 
   const [{ loading: saving }, saveResponse] = useAsyncFn(async () => {
+    if (requireComments.length > 0) {
+      if (comments.length === 0 && isLinkedBoxChecked()) {
+        alertApi.post({
+          message:
+            'The selected option(s) require a comment. Please provide a comment.',
+          severity: 'info',
+        });
+        return;
+      }
+      if (comments.length > 0 && !isLinkedBoxChecked()) {
+        alertApi.post({
+          message: 'Please select the option(s) that require a comment.',
```

Comment:
perhaps you can move this block outside `useAsyncFn` in another function:

```ts
const handleSaveResponse = () => {
    if (requireComments.length > 0) {
      if (comments.length === 0 && isLinkedBoxChecked()) {
        alertApi.post({
          message:
            'The selected option(s) require a comment. Please provide a comment.',
          severity: 'info',
        });
        return;
      }
      if(....) { }
      saveResponse();
};
```

otherwise you need to add `requireComments`, `alertApi` and `isLinkedBoxChecked` as dependencies (without forgetting to wrap `isLinkedBoxChecked` in a `useMemo`)

Reply:
I removed these changes as a result of feedback received from @kuangp